src/posthog-core.js

```diff
@@ -798,9 +798,13 @@ PostHogLib.prototype.identify = function (new_distinct_id, userProperties) {
         this.register({ distinct_id: new_distinct_id })
     }
 
-    // send an $identify event any time the distinct_id is changing - logic on the server
-    // will determine whether or not to do anything with it.
-    if (new_distinct_id !== previous_distinct_id) {
+    // send an $identify event any time the distinct_id is changing and the old ID is an anoymous ID
+    // - logic on the server will determine whether or not to do anything with it.
+    if (
+        new_distinct_id !== previous_distinct_id &&
+        (!this['persistence']?.props?.$device_id || previous_distinct_id === this['persistence']?.props?.$device_id)
```

Comment:
`?.` will cause errors on older browsers, we have not yet configured babel to handle it.

I think the right way to handle this is via `this.get_property('$device_id')`